src/gtest/test_keystore.cpp

```diff
@@ -43,7 +43,64 @@ TEST(keystore_tests, store_and_retrieve_note_decryptor) {
 
     keyStore.AddSpendingKey(sk);
     EXPECT_TRUE(keyStore.GetNoteDecryptor(addr, decOut));
-    EXPECT_EQ(ZCNoteDecryption(sk.viewing_key()), decOut);
+    EXPECT_EQ(ZCNoteDecryption(sk.receiving_key()), decOut);
+}
+
+TEST(keystore_tests, StoreAndRetrieveViewingKey) {
+    CBasicKeyStore keyStore;
+    libzcash::ViewingKey vkOut;
+    libzcash::SpendingKey skOut;
+    ZCNoteDecryption decOut;
+
+    auto sk = libzcash::SpendingKey::random();
+    auto vk = sk.viewing_key();
+    auto addr = sk.address();
+
+    // Sanity-check: we can't get a viewing key we haven't added
+    EXPECT_FALSE(keyStore.HaveViewingKey(addr));
+    EXPECT_FALSE(keyStore.GetViewingKey(addr, vkOut));
+
+    // and we shouldn't have a spending key or decryptor either
+    EXPECT_FALSE(keyStore.HaveSpendingKey(addr));
+    EXPECT_FALSE(keyStore.GetSpendingKey(addr, skOut));
+    EXPECT_FALSE(keyStore.GetNoteDecryptor(addr, decOut));
+
+    // and we can't find it in our list of addresses
+    std::set<libzcash::PaymentAddress> addresses;
+    keyStore.GetPaymentAddresses(addresses);
+    EXPECT_FALSE(addresses.count(addr));
```

Comment:
Is `EXPECT_EQ(0, ... ` better?  What if false condition is not 0 on some system?  Hypothetical of course!

Reply:
This pattern is used all over the place, both in Boost tests and gtests. If 0 != false on some system, we'll have bigger problems :upside_down_face: